myriad-scheduler/src/main/java/org/apache/myriad/configuration/MyriadConfiguration.java

```diff
@@ -197,47 +200,44 @@ public String getFrameworkRole() {
     return nmInstances;
   }
 
-  public Boolean isRebalancer() {
-    return rebalancer != null ? rebalancer : DEFAULT_REBALANCER;
+  public Boolean isRebalancerEnabled() {
+    return Optional.of(rebalancerEnabled).or(DEFAULT_REBALANCER_ENABLED);
   }
 
   public Boolean isHAEnabled() {
-    return haEnabled != null ? haEnabled : DEFAULT_HA_ENABLED;
+    return Optional.of(haEnabled).or(DEFAULT_HA_ENABLED);
   }
 
   public NodeManagerConfiguration getNodeManagerConfiguration() {
-    return this.nodemanager;
+    return nodeManager;
   }
 
-  public Map<String, ServiceConfiguration> getServiceConfigurations() {
-    return this.services;
+  public Optional<Map<String, ServiceConfiguration>> getServiceConfigurations() {
+    return Optional.fromNullable(services);
   }
```

Comment:
Yep, makes sense, I will update accordingly.

--John

On Wed, May 11, 2016 at 10:59 PM, Darin <notifications@github.com> wrote:

> In
> myriad-scheduler/src/main/java/org/apache/myriad/configuration/MyriadConfiguration.java
> <https://github.com/apache/incubator-myriad/pull/71#discussion_r62958979>:
>
> >    }
> >
> > -  public Map<String, ServiceConfiguration> getServiceConfigurations() {
> > -    return this.services;
> > +  public Optional<Map<String, ServiceConfiguration>> getServiceConfigurations() {
> > +    return Optional.fromNullable(services);
> >    }
>
> Seems like just sending an empty Map here would avoid all the null and
> isPresent checks later. Thoughts?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-myriad/pull/71/files/9bb1e6b7514d58b5cb128bd3e23bec0a03f55aaa#r62958979>
>